hrm-service/src/contact-job/contact-job-complete.ts

```diff
@@ -44,18 +48,34 @@ export const pollAndprocessCompletedContactJobs = async () => {
       try {
         const completedJob: CompletedContactJobBody = JSON.parse(m.Body);
 
-        await processCompletedContactJob(completedJob);
+        if (completedJob.attemptResult === 'success') {
+          await processCompletedContactJob(completedJob);
+
+          // Mark the job as completed
+          const markedComplete = await completeContactJob(
+            completedJob.jobId,
+            completedJob.attemptPayload,
+          );
+
+          // Delete the message from the queue (this could be batched)
+          await deletedCompletedContactJobs(m.ReceiptHandle);
+
+          return markedComplete;
+        } else {
+          const { jobId, attemptNumber, attemptPayload } = completedJob;
+          const updated = await appendFailedAttemptPayload(jobId, attemptNumber, attemptPayload);
 
-        // Mark the job as completed
-        const markedComplete = await completeContactJob(
-          completedJob.jobId,
-          completedJob.completionPayload,
-        );
+          if (attemptNumber === jobMaxAttempts) {
+            const markedComplete = await completeContactJob(
+              completedJob.jobId,
+              'Attempts limit reached',
```

Comment:
Can we make this a JSON object? Even if it's just `{ "message": "Attempts limit reached" }`

It's harder to manage versioning going from a string to an object if we want to add more info later than it is just to add more properties to an existing object